### PR TITLE
Introducing BulkOptions.

### PR DESCRIPTION
**Pull request by sduskis**
It encapsulates the options for bulk operations that were previously in BigtableOptions and AbstractBigtableConnection.

### CODE REVIEW

File: bigtable-client-core/src/main/java/com/google/cloud/bigtable/config/BigtableOptions.java [hunk 1]
```diff
@@ -205,6 +167,11 @@ public Builder setUsePlaintextNegotiation(boolean usePlaintextNegotiation) {
     }
 
     public BigtableOptions build() {
+      if (bulkOptions == null) {
+        int maxInflightRpcs =
```
Comment (garye) [inline bigtable-client-core/src/main/java/com/google/cloud/bigtable/config/BigtableOptions.java]: Is it possible in a non-HBase environment to have a non-null bulkOptions but to not have set maxInflightRpcs? If so we might not have a valid value to pass through to ResourceLimiter.

**Comment by garye** [comment]
LGTM other than the one concern

**Comment by sduskis** [comment]
I pushed a commit that addresses the concern.  PTAL

**Comment by garye** [comment]
LGTM

**Pull request merged**